code/modules/library/soapstone.dm

```diff
@@ -0,0 +1,127 @@
+/obj/item/soapstone
+    name = "soapstone"
+    desc = "Leave informative messages for the crew, including the crew of future shifts!\n(Disclaimer: Does not work on shuttles)\n WARNING: Side effects may include beatings, bannings and orbital bombardment."
+    icon = 'icons/obj/items.dmi'
+    icon_state = "soapstone"
+    throw_speed = 3
+    throw_range = 5
+    w_class = WEIGHT_CLASS_TINY
+    var/tool_speed = 50
+    var/remaining_uses = 3
+
+
+/obj/item/soapstone/examine(mob/user)
+    . = ..()
+    to_chat(usr, "It has [remaining_uses] uses left.")
+
+/obj/item/soapstone/afterattack(atom/target, mob/user, proximity)
+    var/turf/T = get_turf(target)
+    if(!proximity)
+        return
+
+    if(!remaining_uses)
+        // The dull chisel is dull.
+        to_chat(usr, "<span class='warning'>[src] is dull It cannot engrave.</span>")
+
+    var/obj/structure/chisel_message/already_message = locate(/obj/structure/chisel_message) in T
+
+    if(!good_chisel_message_location(T))
+        to_chat(usr, "<span class='warning'>It's not an appropriate location to engrave on [T].</span>")

```

Comment:
`user` argument here (and in the rest of this proc). Also this sentence seems awkward though not sure what you would change it to.

Reply:
I had an idea of like `The stone wont engrave onto this location` but couldnt think